src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurCashAdvancePdpFeedFileServiceImpl.java

```diff
@@ -51,31 +53,42 @@ public boolean createPdpFeedFileForValidatedDetailFileLines(ConcurRequestExtract
         return pdpFileSuccessfullyCreated;
     }
 
-    private PdpFeedFileBaseEntry buildPdpFeedBaseEntry(ConcurRequestExtractFile requestExtractFile) {
-        int totalPdpDetailRecordsCount = 0;
-        KualiDecimal totalPdpDetailRecordsAmount = KualiDecimal.ZERO;
+    private PdpFeedFileBaseEntry buildPdpFeedBaseEntry(ConcurRequestExtractFile requestExtractFile, ConcurRequestExtractBatchReportData reportData) {
         PdpFeedFileBaseEntry pdpBaseEntry = new PdpFeedFileBaseEntry();
         pdpBaseEntry.setHeader(buildPdpFeedHeaderEntry(requestExtractFile.getBatchDate()));
         List<PdpFeedGroupEntry> groupEntries = new ArrayList<PdpFeedGroupEntry>();
+
         for (ConcurRequestExtractRequestDetailFileLine detailFileLine : requestExtractFile.getRequestDetails()) {
             if (detailFileLine.getValidationResult().isCashAdvanceLine() && detailFileLine.getValidationResult().isValidCashAdvanceLine()) {
-              PdpFeedDetailEntry pdpDetailEntry = buildPdpFeedDetailEntry(detailFileLine, buildPdpFeedAccountingEntry(detailFileLine));
-              List<PdpFeedDetailEntry> pdpDetailEntries = new ArrayList<PdpFeedDetailEntry>();
-              pdpDetailEntries.add(pdpDetailEntry);
-
-              groupEntries.add(buildPdpFeedGroupEntry(detailFileLine, buildPdpFeedPayeeIdEntry(detailFileLine), pdpDetailEntries));
-
-              recordCashAdvanceGenerationInDuplicateTrackingTable(detailFileLine, pdpDetailEntry.getSourceDocNbr(), requestExtractFile.getFileName());
-
-              totalPdpDetailRecordsCount++;
-              totalPdpDetailRecordsAmount = totalPdpDetailRecordsAmount.add(detailFileLine.getRequestAmount());
-          }
-          else if (detailFileLine.getValidationResult().isCashAdvanceLine() && detailFileLine.getValidationResult().isNotValidCashAdvanceLine()) {
-              LOG.info("buildPdpFeedBaseEntry: Cash Advance was detected but validation failed for:  " + KFSConstants.NEWLINE + detailFileLine.toString());
-          }
+                PdpFeedDetailEntry pdpDetailEntry = buildPdpFeedDetailEntry(detailFileLine, buildPdpFeedAccountingEntry(detailFileLine));
```

Comment:
That is what I was attempting before which the reviewers did not like because of the local variables.  The state of each line of data from the file is held in file line's attributes.  Our options are:

(a) Have two separate methods with the exact same for-lopp and if-else-if structure that goes through all of the data lines of the file.  One would build the pdpFeedsBaseEntry and the other method would build the data for the report. I do not like this as we would not be actually reporting on the processing that has happened to create the pdpFeedsBaseEntry. We would be recreating the report data asynchronously afterward.

(b) Put all of the local counters back in and call a method that would set all the report data after the pdpFeedsBaseEntry was built.  Previous review comments indicated that was not an acceptable clean code approach but it gives you the counters for the processing that are separate from the reporting where a separate method can be called to generate the report data. This is my preference.

(c) Leaving it as. Not my preference either since those two things are are separate.

Let me know how to proceed.